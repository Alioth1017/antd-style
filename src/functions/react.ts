import _styled from '@emotion/styled';

import type { CreateStyled } from '@/types';

export const styled = _styled as CreateStyled;
export { css as reactCss, ThemeProvider as EmotionThemeProvider, withTheme } from '@emotion/react';
