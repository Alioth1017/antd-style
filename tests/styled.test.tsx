import { render } from '@testing-library/react';

import { styled, ThemeProvider } from 'antd-style';

describe('styled', () => {
  it('类型定义正常', () => {
    const App = styled.div`
      background: ${(p) => p.theme.colorPrimary};
    `;

    expect(App).toBeTruthy();
  });
  describe('使用 antd 主题样式', () => {
    it('没有包裹 Provider 的情况下，找不到样式', () => {
      const App = styled.div`
        width: 30px;
        color: ${(p) => p.theme.colorPrimary};
        background: ${(p) => p.theme.colorPrimaryBg};
      `;

      const { container } = render(<App>自定义样式</App>);
      expect(container).toHaveStyle({ color: undefined, background: undefined, width: 30 });
      expect(container).toMatchSnapshot();
    });

    it('在 Provider 包裹下，能正常获得样式', () => {
      const App = styled.div`
        width: 30px;
        color: ${(p) => p.theme.colorPrimary};
        background: ${(p) => p.theme.colorPrimaryBg};
      `;

      const { container } = render(
        <ThemeProvider>
          <App>自定义样式</App>
        </ThemeProvider>,
      );
      expect(container.firstChild).toHaveStyle({
        width: '30px',
        color: '#1677ff',
        background: '#e6f4ff',
      });

      expect(container).toMatchSnapshot();
    });
  });
});
