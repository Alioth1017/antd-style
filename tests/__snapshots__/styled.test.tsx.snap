// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled 使用 antd 主题样式 在 Provider 包裹下，能正常获得样式 1`] = `
.emotion-0 {
  width: 30px;
  color: #1677ff;
  background: #e6f4ff;
}

<div>
  <div
    class="emotion-0"
  >
    自定义样式
  </div>
</div>
`;

exports[`styled 使用 antd 主题样式 没有包裹 Provider 的情况下，找不到样式 1`] = `
.emotion-0 {
  width: 30px;
}

<div>
  <div
    class="emotion-0"
  >
    自定义样式
  </div>
</div>
`;
